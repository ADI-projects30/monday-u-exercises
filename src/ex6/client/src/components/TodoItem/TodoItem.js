import { React } from "react";
import { ItemClient } from "../../services/item_client";
import PropTypes from "prop-types";
import delete_icon from "../../images/delete_icon.svg";
import Button from "../Button/Button";
import "../TodoItem/TodoItem.css";
import "../Button/Button.css";

const TodoItem = ({ setTodos, todo }) => {
  let itemClient = new ItemClient();

  const deleteHandler = async () => {
    await itemClient.deleteItem(todo);
    setTodos(await itemClient.getItems());
  };

  const toggleHandler = async () => {
    await itemClient.toggleDone(todo);
    setTodos(await itemClient.getItems());
  };

  return (
    <li className="list-item">
      <Button
        as="input"
        id={todo.id}
        status={todo.status}
        change={toggleHandler}
      ></Button>
      <p className="list-item-text">{todo.name}</p>
      <Button as="img" delete_icon={delete_icon} click={deleteHandler}></Button>
    </li>
  );
};

export default TodoItem;

TodoItem.prototypes = {
  setTodos: PropTypes.array,
  todo: PropTypes.string,
};

TodoItem.defaultProps = {
  setTodos: [],
};
