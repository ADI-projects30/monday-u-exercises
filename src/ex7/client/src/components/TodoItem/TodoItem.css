  .list-item-text {
    padding-left: 10px;
  }
  