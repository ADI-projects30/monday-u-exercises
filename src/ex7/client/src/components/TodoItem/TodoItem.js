import { React } from "react";
import PropTypes from "prop-types";
import delete_icon from "../../images/delete_icon.svg";
import Button from "../Button/Button";
import Checkbox from "../Checkbox/Checkbox";
import "../TodoItem/TodoItem.css";
import "../Button/Button.css";
import "../Checkbox/Checkbox.css";

const TodoItem = ({
  name,
  id,
  status,
  handleTodoStatusUpdate,
  handleTodoDelete,
}) => {
  return (
    <li className="list-item">
      <Checkbox
        type="input"
        id={id}
        status={status}
        change={handleTodoStatusUpdate}
      ></Checkbox>
      <p className="list-item-text">{name}</p>
      <Button
        type="img"
        delete_icon={delete_icon}
        click={handleTodoDelete}
      ></Button>
    </li>
  );
};

export default TodoItem;

TodoItem.propTypes = {
  name: PropTypes.string,
  id: PropTypes.number,
  status: PropTypes.oneOfType([PropTypes.number, PropTypes.bool]),
  HandleTodoUpdate: PropTypes.func,
  handleTodoDelete: PropTypes.func,
};

TodoItem.defaultProps = {
  status: 0,
};
