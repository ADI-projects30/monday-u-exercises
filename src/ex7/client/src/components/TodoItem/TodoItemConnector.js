import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { getItemsList } from "../../selectors/items-entities-selectors";
import {
  loadtodosListAction,
  deleteTodo,
  toggleCompleteTodoAction,
} from "../../actions/todo-actions";
import TodoItem from "./TodoItem";
// import TodoItem from "./TodoItem";

const mapStateToProps = (state, ownProps) => {
  const todosList = getItemsList(state);
  // const isLoggedIn = getIsLoggedIn(state);
  return { todosList };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return bindActionCreators(
    { loadtodosListAction, deleteTodo, toggleCompleteTodoAction },
    dispatch
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(TodoItem);
