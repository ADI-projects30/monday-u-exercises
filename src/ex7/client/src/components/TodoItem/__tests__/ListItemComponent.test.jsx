import renderer from "react-test-renderer";
import '@testing-library/jest-dom';
import TodoItem from "../TodoItem";
import React from "react";
import { Provider } from "react-redux";
import { store } from "../../../store";

describe("Snapshot test ListItem", () => {
    
    const item = {
            id: 33,
            name: "See a movie",
            status: false,
    }

        
    it("renders correctly", () => {
        const tree = renderer.create(
                <Provider store={store}>
                    <TodoItem id={item.id} name={item.name} status={item.status}/>
                </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});


  