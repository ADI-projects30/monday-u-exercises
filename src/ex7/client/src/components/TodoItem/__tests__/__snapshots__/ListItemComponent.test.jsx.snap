// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test ListItem renders correctly 1`] = `
<li
  className="list-item"
>
  <input
    checked={false}
    className="list-item-checkbox"
    id={33}
    name="done"
    type="checkbox"
    value="done"
  />
  <p
    className="list-item-text"
  >
    See a movie
  </p>
  <img
    alt="delete"
    className="list-item-delete-button"
    src="delete_icon.svg"
  />
</li>
`;
